### PR TITLE
Add SR-IOV CNI plugin

### PR DESCRIPTION
**Pull request by hustcat**
Add SR-IOV CNI plugin.  [#netlink](https://github.com/vishvananda/netlink) needed to be bump to the latest.

### CODE REVIEW

File: plugins/main/sriov/README.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
+# SR-IOV CNI plugin
+
```
Comment (steveeJ) [inline plugins/main/sriov/README.md]: It seems a bit empty here! Could you add an introduction to this plugin that explains what it does and what it's used for?

Comment (hustcat) [inline plugins/main/sriov/README.md]: Introduction added:)

File: plugins/main/sriov/README.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+PF is used by host.Each VFs can be treated as a separate physical NIC and assigned to one container, and configured with separate MAC, VLAN and IP, etc.
+
+## Enable SR-IOV
+
+Given Intel ixgbe NIC on CentOS, Fedora or RHEL:
+
+```
+# vi /etc/modprobe.conf
+options ixgbe max_vfs=8,8
```
Comment (steveeJ) [inline plugins/main/sriov/README.md]: this is a requirement for the plugin to work or just an example?

**Comment by steveeJ** [comment]
In general this plugin looks fine and I'm in favor of merging this PR. A couple of notes:

- Please adjust the commit message(s) to conform to https://github.com/containernetworking/cni/blob/master/CONTRIBUTING.md#format-of-the-commit-message, thanks a lot!
- Tests please!
- Nitpick(s) in the docs

File: plugins/main/sriov/README.md (hunk 2 above)
Comment (hustcat) [inline plugins/main/sriov/README.md]: It's requirement. SR-IOV plugin need NIC enable SR-IOV capability, which not enabled by default.

**Comment by hustcat** [comment]
@steveeJ I add some tests, but didn't enable it by default. You can run it with SR-IOV NIC.

**Comment by rosenhouse** [comment]
@steveeJ @hustcat I'm not sure how I feel about us having plugins with tests that we can't actually run in CI.  I think that us maintaining the plugin in the repo implies a certain level of support and endorsement, which I expect to be backed-up by test coverage running in our CI.

Perhaps such plugins may be better maintained by 3rd parties.

One way to address this could be to maintain a list in our readme, pointing to 3rd party plugins living in other repos.  So there can be some level of information sharing and social capital, without us taking on the burden of maintaining those plugins 

Thoughts?  @containernetworking/cni-maintainers

**Comment by hustcat** [comment]
Created a standalone repo [here](https://github.com/hustcat/sriov-cni).

**Comment by rosenhouse** [comment]
Thanks @hustcat I've opened a separate PR https://github.com/containernetworking/cni/pull/272 that adds your standalone repo to the list of 3rd party CNI plugins.

**Comment by rosenhouse** [comment]
@hustcat That README PR has been merged -- so your plugin repo is now linked to from the main page.

I'm going to close this PR now.  Feel free to re-open if you'd like us to re-consider your plugin for merging into this repo, but then I'd ask that we figure out how to run the test suite in our CI.

**Pull request closed without merging**